certora/specs/SocialRecoveryModule.spec

```diff
@@ -502,3 +503,113 @@ rule finalizeRecoveryAlwaysPossible(env e) {
+    uint64 executeAfterBefore = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersBefore = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    f(e, args);
+
+    uint256 guardianApprovalCountAfter = currentContract.recoveryRequests[safeContract].guardiansApprovalCount;
+    uint256 newThresholdAfter = currentContract.recoveryRequests[safeContract].newThreshold;
+    uint64 executeAfterAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersAfter = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    assert (
+        guardianApprovalCountBefore != guardianApprovalCountAfter ||
+        newThresholdBefore != newThresholdAfter ||
+        executeAfterBefore != executeAfterAfter ||
+        newOwnersBefore != newOwnersAfter
+        ) =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector ||
+        f.selector == sig:executeRecovery(address,address[],uint256).selector ||
+        f.selector == sig:finalizeRecovery(address).selector ||
+        f.selector == sig:cancelRecovery().selector;
+}
+
+// This rule verifies that is the confirmedHashes change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+rule confirmedHashesChange(env e, calldataarg args, method f, bytes32 hash, address guardian) {
+    bool confirmedHashBefore = currentContract.confirmedHashes[hash][guardian];
+
+    f(e, args);
+
+    bool confirmedHashAfter = currentContract.confirmedHashes[hash][guardian];
+
+    assert confirmedHashBefore != confirmedHashAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector;
+}
+
+// This rule verifies that is the walletsNonces change, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - invalidateNonce(...)
+rule walletsNoncesChange(env e, calldataarg args, method f) {
+    uint256 walletsNoncesBefore = currentContract.walletsNonces[safeContract];
+
+    f(e, args);
+
+    uint256 walletsNoncesAfter = currentContract.walletsNonces[safeContract];
+
+    assert walletsNoncesBefore != walletsNoncesAfter =>
+        f.selector == sig:confirmRecovery(address,address[],uint256,bool).selector ||
+        f.selector == sig:multiConfirmRecovery(address,address[],uint256,SocialRecoveryModule.SignatureData[],bool).selector ||
+        f.selector == sig:executeRecovery(address,address[],uint256).selector ||
+        f.selector == sig:invalidateNonce().selector;
+}
+
+// This rule verifies that the recovery period never changes.
+rule recoveryPeriodNeverChange(env e, calldataarg args, method f) {
```

Comment:
Idem on env and calldataarg... you get the point :P

Reply:
Done 👍🏾